### ISSUE TITLE
Column dragging breaks block

### ISSUE THREAD

**Issue by evolvesnc** [WordPress/gutenberg#9303]
Hello everyone,

when dragging empty columns, an error is generated which prevents from using the block altogether. Also, the "Attempt recovery" button that allows recovering the block content doesn't seem to be working.

Here's a video that explains how to reproduce the issue: https://www.dropbox.com/s/ohaceh7efsmj1st/aug-23-2018_21-28-06.mp4?dl=0

Browser: Chrome

**Comment by afercia** [comment]
I was able to reproduce this. Interestingly, I get an `Uncaught RangeError: Maximum call stack size exceeded` 😱 

Error copied from the "Attempt recovery" warning:

```
RangeError: Maximum call stack size exceeded
    at map (http://build.wordpress-develop.test/wp-content/plugins/gutenberg/vendor/lodash.2cfc2504.js:9544:17)
    at http://build.wordpress-develop.test/wp-content/plugins/gutenberg/build/editor/index.js?ver=1535298354:12:108560
    at d (http://build.wordpress-develop.test/wp-content/plugins/gutenberg/build/editor/index.js?ver=1535298354:12:24030)
    at http://build.wordpress-develop.test/wp-content/plugins/gutenberg/build/editor/index.js?ver=1535298354:12:109235
    at d (http://build.wordpress-develop.test/wp-content/plugins/gutenberg/build/editor/index.js?ver=1535298354:12:24030)
    at http://build.wordpress-develop.test/wp-content/plugins/gutenberg/build/editor/index.js?ver=1535298354:12:108588
    at arrayMap (http://build.wordpress-develop.test/wp-content/plugins/gutenberg/vendor/lodash.2cfc2504.js:631:23)
    at map (http://build.wordpress-develop.test/wp-content/plugins/gutenberg/vendor/lodash.2cfc2504.js:9546:14)
    at http://build.wordpress-develop.test/wp-content/plugins/gutenberg/build/editor/index.js?ver=1535298354:12:108560
    at d (http://build.wordpress-develop.test/wp-content/plugins/gutenberg/build/editor/index.js?ver=1535298354:12:24030)
```

**Comment by nosolosw** [comment]
For the record: the columns block doesn't need to be void for this error to occur.

I could only repro by dropping the columns block on itself (not happening always, but I can repro almost every time if I drop it on the drag area opposed to which I've used for dragging it - if I drag from the left side, I'd drop on the right side).

**Comment by nosolosw** [comment]
I've isolated the bug to an infinite loop caused between `getBlock` and `getBlockCacheBust`: they call each other until they run out of space in the stack. I've tried a band-aid (avoid using `getBlock` in some situations) but it surfaced the bug elsewhere due to the omnipresent use of `getBlock`.

**Comment by nosolosw** [comment]
This will be fixed by https://github.com/WordPress/gutenberg/pull/10201 which is scheduled for 4.0.

**Comment by nosolosw** [comment]
https://github.com/WordPress/gutenberg/pull/10201 has been merged and the fix will be included in the 4.0 RC.
